src/SqlAsyncCollector.cs

```diff
@@ -212,7 +212,9 @@ private static void GenerateDataQueryForMerge(TableInformation table, IEnumerabl
             }
 
             rowData = JsonConvert.SerializeObject(rowsToUpsert, table.JsonSerializerSettings);
-            newDataQuery = $"WITH {NewDataParameter} AS ( SELECT * FROM OPENJSON({RowDataParameter}) WITH ({string.Join(",", table.ColumnDefinitions)}) )";
+            IEnumerable<string> columnNamesFromPOCO = typeof(T).GetProperties().Select(prop => prop.Name);
```

Comment:
@chlafreniere @Charles-Gagnon I know we chatted about the use of "POCO" before. I'm using it here to be consistent with other parts of the code but let me know if something else would be preferred.